### ISSUE TITLE
<fstream>: Importing as a header unit leads to error C2079: undefined class 'std::basic_ofstream<char,std::char_traits<char>>'

### ISSUE THREAD

**Issue by orikama** [microsoft/STL#3112]
After updating to `Visual Studio 17.4.0 Preview 2.0` using `<fstream>` as a header unit leads to compilation error.

`Text.ixx`
```c++
export module Test;

// Importing <filesystem> and/or <string> with <fstream> leads to compilation error.
// But importing <fstream> without <filsystem> and <string> or importing it first works.
import <filesystem>;
import <string>;
import <fstream>;

export class Test
{
    std::ofstream stream;
};
```

`main.cpp`
```c++
import Test;

int main()
{
    Test t;
}
```

**Command-line test case**
```powershell
D:\test> cl /std:c++20 /EHsc /exportHeader /headerName:angle fstream string filesystem
Microsoft (R) C/C++ Optimizing Compiler Version 19.34.31823.3 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

fstream
string
filesystem

D:\test> cl /c /std:c++20 /EHsc .\Test.ixx .\main.cpp /headerUnit:angle fstream=.\fstream.ifc /headerUnit:angle string=.\string.ifc /headerUnit:angle filesystem=.\filesystem.ifc
Microsoft (R) C/C++ Optimizing Compiler Version 19.34.31823.3 for x64
Copyright (C) Microsoft Corporation.  All rights reserved.

Test.ixx
main.cpp
D:\test\Test.ixx(12): error C2079: 'Test::stream' uses undefined class 'std::basic_ofstream<char,std::char_traits<char>>'
Generating Code...
```

**Comment by fsb4000** [comment]
Yes, I tested and I have the same results.
Might be duplicate of https://developercommunity.visualstudio.com/t/10076720 but I'm not sure...

**Comment by fsb4000** [comment]
I created https://developercommunity.visualstudio.com/t/10154236

**Comment by StephanTLavavej** [comment]
FYI @cdacamar, we believe this is a compiler bug. Thanks @fsb4000 for reporting this on DevCom (where it will create an internal bug for Cameron to use). :heart_eyes_cat:

**Comment by fsb4000** [comment]
"Fixed - Pending Release": https://developercommunity.visualstudio.com/t/10154236

**Comment by CaseyCarter** [comment]
The fix should ship in 17.5 Preview 1 or 2.

**Comment by StephanTLavavej** [comment]
Closing as the compiler fix has been checked in - it should be available in 17.5 Preview 1. Thanks again for reporting this! :smile_cat:
